pipeline/transformer/transformer_sierra/src/test/scala/uk/ac/wellcome/platform/transformer/sierra/fixtures/HybridRecordReceiverFixture.scala

```diff
@@ -1,34 +1,119 @@
+
+  val memoryIndexedStore =
+    new MemoryStore[
+      Version[String, Int],
+      HybridIndexedStoreEntry[String, EmptyMetadata]](Map.empty)
+
+  implicit val memoryStreamStore = MemoryStreamStore[String]()
+
+  implicit val hybridStore =
+    new HybridStore[
+      Version[String, Int],
+      String,
+      SierraTransformable,
+      EmptyMetadata] {
+      override implicit val indexedStore = memoryIndexedStore
+      override implicit val typedStore
+        : MemoryTypedStore[String, SierraTransformable] =
+        new MemoryTypedStore[String, SierraTransformable](Map.empty)
+      override def createTypeStoreId(id: Version[String, Int]): String =
+        s"${id.id}/${id.version}"
+    }
 
-trait HybridRecordReceiverFixture extends Messaging with SNS {
   def withHybridRecordReceiver[R](
     topic: Topic,
     bucket: Bucket,
     snsClient: AmazonSNS = snsClient
-  )(testWith: TestWith[HybridRecordReceiver, R])(
-    implicit objectStore: ObjectStore[SierraTransformable]): R =
-    withMessageWriter[TransformedBaseWork, R](
-      bucket,
-      topic,
-      writerSnsClient = snsClient) { messageWriter =>
-      val recordReceiver = new HybridRecordReceiver(
-        messageWriter = messageWriter,
-        objectStore = objectStore
-      )
-
-      testWith(recordReceiver)
+  )(testWith: TestWith[HybridRecordReceiver[SNSConfig], R]): R =
+    withSqsBigMessageSender[TransformedBaseWork, R](bucket, topic, snsClient) {
+      msgSender =>
+        val recordReceiver = new HybridRecordReceiver(
+          msgSender = msgSender,
+          store = hybridStore
+        )
+
+        testWith(recordReceiver)
     }
+
+  def createHybridRecordNotificationWith(
+    sierraTransformable: SierraTransformable,
+    version: Int = 1): NotificationMessage = {
+
+    val hybridRecord = createHybridRecordWith(
+      sierraTransformable,
+      version = version
+    )
+    createNotificationMessageWith(
+      message = hybridRecord
+    )
+  }
+
+  def createHybridRecordWith(
+    sierraTransformable: SierraTransformable,
+    version: Int = 1,
+    id: String = Random.alphanumeric take 10 mkString): HybridRecord = {
+
+    hybridStore.put(Version(id, version))(
+      HybridStoreEntry(sierraTransformable, EmptyMetadata()))
+    HybridRecord(
+      id = id,
+      version = version,
+      location = ObjectLocation("namespace", "path")
+    )
+  }
+
+  def createCorruptedHybridRecord(
+    version: Int = 1,
+    id: String = Random.alphanumeric take 10 mkString): HybridRecord = {
+
+    val typeId = s"${id}/${version}"
+    val stream = new ByteArrayInputStream("{\"x\": 1}".getBytes)
+    memoryIndexedStore.put(Version(id, version))(
+      HybridIndexedStoreEntry(typeId, EmptyMetadata()))
+    memoryStreamStore.put(typeId)(
+      new InputStreamWithLengthAndMetadata(stream, 8L, Map.empty))
+    HybridRecord(
+      id = id,
+      version = version,
+      location = ObjectLocation("namespace", "path")
```

Comment:
Ditto above.

Reply:
Could just set this location to a val with a comment explaining what / why it is.
val `irrelevantLocation = ObjectLocation("namespace", "path") // blah blah`